### PR TITLE
Adding auth to nuxt app

### PR DESCRIPTION
**Pull request by funkel1989**
## Known Bugs
- The login button is currently cancelling mid request. Not sure why yet. Clicking the go to dashboard button will force you to login though because it is a protected route
- I had to remove nuxt transitions as it was breaking server side hydration... never seen this before either in 6 years working with nuxt, fml

## Considerations
- added in ESLint for nuxt to throw build errors on bad formatting. I think this will help others with formatting
- added in a shim file for deployment when we get to this. its needed for a aws lambda deployment and I didn't want to forget it
- Removed package-lock.json as we are using yarn, not npm

## How to make auth work?
- Create .env file in your root. this file is git ignored and should stay this way
- Reach out to Michael (funkel1989) for your env variables you need to make it work.

### CODE REVIEW

File: app.vue [hunk 1]
```diff
@@ -1,23 +1,10 @@
+<script lang="ts" setup></script>
+
 <template>
-  <div>
-    <h1 class="title">Welcome to Constant Reminders</h1>
-    <p id="electron-status">isElectron: {{ useElectron().isElectron }}</p>
```
Comment (aadamsongit) [inline app.vue:4]: Not sure if we're going to add this back at some point. I know it was throwing an error, before...

**Review by aadamsongit — APPROVED** [review]
I've reviewed the code. Michael answered my questions about some of the changes. Seems he has a good idea of what we'll need to be focusing on, going forward.

File: app.vue (hunk 1 above)
Comment (funkel1989) [reply app.vue:4]: Naw, this was just for debug purposes

**Review by vphillips1 — APPROVED** [review]
LGTM

**Pull request merged**